vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -458,6 +477,16 @@ func (d *VolumeDriver) Unmount(r volume.UnmountRequest) volume.Response {
 		return volume.Response{Err: ""}
 	}
 
+	if plugin_utils.IsFullVolumeName(r.Name) != true {
+		fullVolName := fullVolumeNameMap[r.ID]
+		if fullVolName == "" { // if ID not present in local map, do a get trip
```

Comment:
I think I saw this code a few time already :-). 
Please refactor to avoid cut-n-paste